dix/req.c

```diff
@@ -1365,14 +1365,11 @@ static void dix__rop(struct m0_dix_req *req, const struct m0_bufvec *keys,
 	M0_PRE(req->dr_indices_nr == 1);
 
 	keys_nr = keys->ov_vec.v_nr;
-#ifndef DTM0
-	M0_PRE(keys_nr != 0);
-#else
+	M0_PRE(ergo(!ENABLE_DTM0, keys_nr != 0));
 	/* We support only one record per request in DTM0. */
-	M0_PRE(ergo(M0_IN(req->dr_type, (DIX_CREATE, DIX_DELETE,
-					 DIX_PUT, DIX_DEL)),
+	M0_PRE(ergo(ENABLE_DTM0 && M0_IN(req->dr_type, (DIX_CREATE, DIX_DELETE,
+							DIX_PUT, DIX_DEL)),
```

Comment:
Propose to use the following condition
=====================================
	/* We support only one record per request in DTM0. */
	M0_PRE(ergo(req->dr_dtx != NULL && M0_IN(req->dr_type,
						 (DIX_CREATE, DIX_DELETE,
						  DIX_PUT, DIX_DEL)),
		    keys_nr == 1));
=====================================
as we don't have any restrictions for the operations that go without dtx.